From 6646c0116fd8ce2a8a81c10978f59ecd1db87917 Mon Sep 17 00:00:00 2001
From: Gareth Stockwell <gareth.stockwell@accenture.com>
Date: Fri, 7 Oct 2011 11:39:48 +0100
Subject: [PATCH 2/4] Fix transition to QMediaPlayer::NoMedia mediaStatus

When source is set to a null string, QMediaPlayer::mediaStatus should
transition to QMediPlayer::NoMedia.

Reviewed-by: PENDING
---
 .../mmf/mediaplayer/s60mediaplayercontrol.cpp      |   19 +++++--------------
 .../mmf/mediaplayer/s60mediaplayersession.cpp      |    2 +-
 2 files changed, 6 insertions(+), 15 deletions(-)

diff --git a/plugins/multimedia/symbian/mmf/mediaplayer/s60mediaplayercontrol.cpp b/plugins/multimedia/symbian/mmf/mediaplayer/s60mediaplayercontrol.cpp
index 7807c23..5b38d77 100644
--- a/plugins/multimedia/symbian/mmf/mediaplayer/s60mediaplayercontrol.cpp
+++ b/plugins/multimedia/symbian/mmf/mediaplayer/s60mediaplayercontrol.cpp
@@ -352,27 +352,18 @@ void S60MediaPlayerControl::setMedia(const QMediaContent &source, QIODevice *str
     // store to variable as session is created based on the content type.
     m_currentResource = source;
     S60MediaPlayerSession *newSession = m_mediaPlayerResolver.PlayerSession();
-    m_mediaSettings.setMediaStatus(QMediaPlayer::UnknownMediaStatus);
 
     if (m_session)
         m_session->reset();
-    else {
-        emit mediaStatusChanged(QMediaPlayer::UnknownMediaStatus);
-        emit error(QMediaPlayer::NoError, QString());
-    }
 
     m_session = newSession;
 
-    if (m_session)
+    if (m_session && !source.canonicalUrl().isEmpty())
         m_session->load(source);
-    else {
-        QMediaPlayer::MediaStatus status = (source.canonicalUrl().isEmpty()) ? QMediaPlayer::NoMedia : QMediaPlayer::InvalidMedia;
-        m_mediaSettings.setMediaStatus(status);
-        emit stateChanged(QMediaPlayer::StoppedState);
-        emit error((source.isNull()) ? QMediaPlayer::NoError : QMediaPlayer::ResourceError, 
-                   (source.isNull()) ? "" : tr("Media couldn't be resolved"));
-        emit mediaStatusChanged(status);
-    }
+
+    QMediaPlayer::MediaStatus status = m_session ? m_session->mediaStatus() : QMediaPlayer::NoMedia;
+    m_mediaSettings.setMediaStatus(status);
+
     emit mediaChanged(m_currentResource);
 }
 
diff --git a/plugins/multimedia/symbian/mmf/mediaplayer/s60mediaplayersession.cpp b/plugins/multimedia/symbian/mmf/mediaplayer/s60mediaplayersession.cpp
index b004ffd..76a4fca 100644
--- a/plugins/multimedia/symbian/mmf/mediaplayer/s60mediaplayersession.cpp
+++ b/plugins/multimedia/symbian/mmf/mediaplayer/s60mediaplayersession.cpp
@@ -316,7 +316,7 @@ void S60MediaPlayerSession::reset()
     doStop();
     doClose();
     setState(QMediaPlayer::StoppedState);
-    setMediaStatus(QMediaPlayer::UnknownMediaStatus);
+    setMediaStatus(QMediaPlayer::NoMedia);
     setPosition(0);
 }
 
-- 
1.7.4.msysgit.0